packages/neon/neon/lib/src/pages/login_qrcode.dart

```diff
@@ -0,0 +1,105 @@
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_zxing/flutter_zxing.dart';
+import 'package:neon/src/router.dart';
+import 'package:neon/src/utils/exceptions.dart';
+import 'package:neon/src/utils/login_qrcode.dart';
+import 'package:neon/src/widgets/exception.dart';
+
+class LoginQrcodePage extends StatefulWidget {
+  const LoginQrcodePage({
+    super.key,
+  });
+
+  @override
+  State<LoginQrcodePage> createState() => _LoginQrcodePageState();
+}
+
+class _LoginQrcodePageState extends State<LoginQrcodePage> {
+  String? _lastErrorURL;
+
+  @override
+  Widget build(final BuildContext context) => Scaffold(
+        appBar: AppBar(),
+        body: ReaderWidget(
+          showGallery: false,
+          showToggleCamera: false,
+          showScannerOverlay: false,
+          tryHarder: true,
+          cropPercent: 0,
+          scanDelaySuccess: const Duration(seconds: 3),
+          onScan: (final code) async {
+            String? url;
+            try {
+              url = code.text;
+              if (url == null) {
+                throw InvalidQrcodeException();
+              }
+              final match = parseLoginQrcodeURL(url);
+              if (match == null) {
+                throw InvalidQrcodeException();
+              }
+              await processLoginQrcode(context, match);
+            } catch (e, s) {
+              if (_lastErrorURL != url) {
+                debugPrint(e.toString());
+                debugPrint(s.toString());
+
+                _lastErrorURL = url;
+                NeonException.showSnackbar(context, e);
+              }
+            }
+          },
+        ),
+      );
+}
+
+class LoginQrcodeIntermediatePage extends StatefulWidget {
+  const LoginQrcodeIntermediatePage({
+    required this.serverURL,
+    required this.loginName,
+    required this.password,
+    super.key,
+  });
+
+  final String serverURL;
+  final String loginName;
+  final String password;
+
+  @override
+  State<LoginQrcodeIntermediatePage> createState() => _LoginQrcodeIntermediatePageState();
+}
+
+class _LoginQrcodeIntermediatePageState extends State<LoginQrcodeIntermediatePage> {
+  @override
+  void initState() {
+    super.initState();
+    WidgetsBinding.instance.addPostFrameCallback((final _) {
+      unawaited(
+        processLoginQrcode(
+          context,
+          LoginQrcode(
+            server: widget.serverURL,
+            user: widget.loginName,
+            password: widget.password,
+          ),
+        ),
+      );
+    });
+  }
+
+  @override
+  Widget build(final BuildContext context) => const SizedBox();
+}
+
+Future processLoginQrcode(final BuildContext context, final LoginQrcode qrcode) async {
```

Comment:
If we manage to make the redirection work in the route this should be moved to into the remaining `LoginQrcodePage`